From 86aa649f7abed38aabf8b452f9919d329ae6c684 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty <etaash.mathamsetty@gmail.com>
Date: Sun, 15 Jun 2025 20:25:54 -0400
Subject: [PATCH 097/378] kernel32: Implement timeGetTime.

Cheat engine's speed hack expects this function to exist in kernel32 for some reason
---
 dlls/kernel32/kernel32.spec | 1 +
 dlls/kernel32/sync.c        | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/dlls/kernel32/kernel32.spec b/dlls/kernel32/kernel32.spec
index f0830ebe4ed..2e51fa9b294 100644
--- a/dlls/kernel32/kernel32.spec
+++ b/dlls/kernel32/kernel32.spec
@@ -1717,6 +1717,7 @@
 @ stdcall -import lstrlen(str)
 @ stdcall -import lstrlenA(str)
 @ stdcall -import lstrlenW(wstr)
+@ stdcall timeGetTime()
 
 ################################################################
 # Wine internal extensions
diff --git a/dlls/kernel32/sync.c b/dlls/kernel32/sync.c
index 44fd01002d4..e4f22b73129 100644
--- a/dlls/kernel32/sync.c
+++ b/dlls/kernel32/sync.c
@@ -103,6 +103,14 @@ DWORD WINAPI DECLSPEC_HOTPATCH GetTickCount(void)
     return user_shared_data->TickCount.LowPart;
 }
 
+/***********************************************************************
+ *           timeGetTime       (KERNEL32.@)
+ */
+DWORD WINAPI DECLSPEC_HOTPATCH timeGetTime(void)
+{
+    return user_shared_data->TickCount.LowPart;
+}
+
 /***********************************************************************
  *           RegisterWaitForSingleObject   (KERNEL32.@)
  */
-- 
2.51.0

